vm/src/builtins/bytearray.rs

```diff
@@ -373,9 +375,12 @@ impl PyByteArray {
         self.inner().hex(sep, bytes_per_sep, vm)
     }
 
-    #[pymethod]
-    fn fromhex(string: PyStrRef, vm: &VirtualMachine) -> PyResult<PyByteArray> {
-        Ok(PyBytesInner::fromhex(string.as_str(), vm)?.into())
+    #[pyclassmethod]
+    fn fromhex(cls: PyTypeRef, string: PyStrRef, vm: &VirtualMachine) -> PyResult {
+        let bytes = PyBytesInner::fromhex(string.as_str(), vm)?;
+        let bytes = vm.ctx.new_bytes(bytes);
+        let args = FuncArgs::from(vec![bytes]);
+        Callable::call(&cls, args, vm)
```

Comment:
```suggestion
        Callable::call(&cls, vec![bytes].into(), vm)
```
Oh, I didn't look in the usage very well. maybe this expression also works